src/core/utils.js

```diff
@@ -354,9 +355,14 @@ export function isCompatibleWithUserAgent({
     // confused by this as tofumatt was.
     // See: https://github.com/mozilla/addons-frontend/issues/2074#issuecomment-286983423
     if (maxVersion && mozCompare(browser.version, maxVersion) === 1) {
-      _log.info(oneLine`maxVersion ${maxVersion} for add-on lower than browser
-        version ${browser.version}, but add-on still marked as compatible
-        because we largely ignore maxVersion. See:
+      // See: https://github.com/mozilla/addons-server/issues/5277
+      if (addon.current_version.is_strict_compatibility_enabled) {
+        return { compatible: false, reason: INCOMPATIBLE_LEGACY_EXTENSION };
```

Comment:
Ah okay, I wasn't aware. The compatibility rules are all a bit dizzying. I'll modify the message, cheers 👍